src/utilities/getCurrentDate.ts

```diff
@@ -69,17 +73,35 @@ export function generateDateOrig(currentDate: DateData): Date {
 }
 
 export function generateWeekList(currentDate: DateData): WeekDay[] {
+	const currentDateCopy: DateData = {
+		year: currentDate.year,
+		month: currentDate.month,
+		day: currentDate.day,
+		timestamp: currentDate.timestamp,
+		dateString: currentDate.dateString
+	};
 	const weekList: WeekDay[] = [];
-	const generatedDate = generateDateOrig(currentDate);
+	const generatedDate = generateDateOrig(currentDateCopy);
+
 
 	if (generatedDate.getDay() === 0) {
-		weekList.push({ day: grabWeekDay(0), date: currentDate.day, month: currentDate.month, year: currentDate.year, dateString: getDateString(currentDate) });
+		weekList.push({ day: grabWeekDay(0), date: currentDateCopy.day, month: currentDateCopy.month, year: currentDateCopy.year, dateString: getDateString(currentDateCopy) });
 		for (let i=1; i<7; i++) {
-			weekList[i] = { day: grabWeekDay(i), date: grabDay(currentDate, currentDate.day, currentDate.month, currentDate.year), month: currentDate.month, year: currentDate.year, dateString: getDateString(currentDate) };
+			weekList[i] = { day: grabWeekDay(i), date: grabDay(currentDateCopy, currentDateCopy.day, currentDateCopy.month, currentDateCopy.year), month: currentDateCopy.month, year: currentDateCopy.year, dateString: getDateString(currentDateCopy) };
 		}
 	} else {
-		const prevSunday = findPrevSunday(generatedDate.getDay(), currentDate);
-		
+		const lowerBound = generatedDate.getDay();
+
+		let prevSunday: DateData = {
+			year: currentDateCopy.year,
+			month: currentDateCopy.month,
+			day: currentDateCopy.day,
+			timestamp: currentDateCopy.timestamp,
+			dateString: currentDateCopy.dateString
+		};
+		for (let i=0; i<lowerBound; i++){
+			prevSunday = generatePrevDate(prevSunday, prevSunday.day, prevSunday.month, prevSunday.year);
+		}
 		weekList.push({ day: grabWeekDay(0), date: prevSunday.day, month: prevSunday.month, year: prevSunday.year, dateString: getDateString(prevSunday) });
```

Comment:
No clone needed here. Want prevSunday to change values on grabDay() call to move onto the next date properly.